core/sodasql/common/config_helper.py

```diff
@@ -0,0 +1,121 @@
+    __instance = None
+    __config: Dict = None
+    file_system = FileSystemSingleton.INSTANCE
+
+    @staticmethod
+    def get_instance(path: Optional[str] = None):
+        if ConfigHelper.__instance == None:
+            ConfigHelper()
+        return ConfigHelper.__instance
+
+    def __init__(self, path: Optional[str] = None):
+        if ConfigHelper.__instance != None:
+            raise Exception("This class is a singleton!")
+        else:
+            ConfigHelper.__instance = self
+
+        if path:
+            self.LOAD_PATHS.insert(0, path)
+
+        self.__config = self.config
+
+        if not self.__config:
+            self.init_config_file()
+
+        self.__ensure_basic_config()
+
+    @property
+    def config_path(self) -> str:
+        return self.LOAD_PATHS[0]
+
+    @property
+    def config(self) -> Dict:
+        if not self.__config:
+            self.reload_config()
+
+        return self.__config
+
+    def reload_config(self) -> Dict:
+        for path in self.LOAD_PATHS:
+            logger.info(f"Trying to load Soda Config file {path}.")
+
+            if self.file_system.file_exists(path):
+                self.__config = yaml.load(
+                    self.file_system.file_read_as_str(path),
+                    Loader=yaml.SafeLoader
+                )
+                break
+
+    def get_value(self, key: str):
+        """Get value from loaded config."""
+        return self.config.get(key, None)
+
+    def init_config_file(self) -> None:
+        """Init default config file if not present."""
+        destination = self.config_path
+
+        if self.file_system.file_exists(destination):
+            logger.info(f"Config file {destination} already exists")
+        else:
+            logger.info(f"Creating config YAML file {destination} ...")
+            self.file_system.mkdirs(self.file_system.dirname(destination))
+            self.upsert_config_file(self.DEFAULT_CONFIG)
+
+    def upsert_value(self, key: str, value: str):
+        """Update or insert a value in the config file and refresh loaded state."""
+        config = self.config
+        config[key] = value
+        self.upsert_config_file(config)
+        self.reload_config()
+
+    def upsert_config_file(self, config: Dict):
+        """Write provided config into a yaml file."""
+        self.file_system.file_write_from_str(
+            self.config_path,
+            yaml.dump(
+                config,
+                default_flow_style=False,
+                sort_keys=False
+            )
+        )
+
+    def generate_user_cookie_id(self) -> str:
+        return str(uuid.uuid4())
+
+    def __ensure_basic_config(self) -> None:
+        for key, value in self.DEFAULT_CONFIG.items():
+            if key not in self.config:
+                self.upsert_value(key, value)
+
+    @property
+    def skip_telemetry(self) -> bool:
+        return self.config.get("skip_telemetry", False)
```

Comment:
just a naming question here. I would probably go for a different name such as `send_anonymous_usage_stats` with a True default. I think from a user perspective it's more friendly to ask them for a positive question rather than negative. That does mean reversing our boolean logic but shouldn't be too complex.

This also is totally identical to the dbt approach (https://docs.getdbt.com/reference/profiles.yml#send_anonymous_usage_stats) and should be more familiar to users.